flexy-pool-core/src/main/java/com/vladmihalcea/flexypool/FlexyPoolDataSource.java

```diff
@@ -225,10 +230,23 @@ public FlexyPoolDataSource() {
      * @param targetDataSource target {@link DataSource}
      */
     public FlexyPoolDataSource(T targetDataSource) {
-        this(new ConfigurationLoader<T>(targetDataSource).getFlexyPoolDataSourceConfiguration());
+        this(new ConfigurationLoader<T>(targetDataSource, null).getFlexyPoolDataSourceConfiguration());
     }
 
     /**
+     * Initialize <code>FlexyPoolDataSource</code> from declarative properties configuration and using the given
+     * target {@link DataSource}. Will pick first properties passed in props. These are
+     * typically properties defined in JPA persistence.xml file.
+     *
+     * @param targetDataSource target {@link DataSource}
+     */
+    public FlexyPoolDataSource(T targetDataSource, Map props) {

```

Comment:
Are you going to pass the properties using the programmatic configuration? The declarative programmatic approach was designed for Java EE. The programmatic one is more suitable for Spring.